src/Microsoft.AspNetCore.SignalR/HubEndPoint.cs

```diff
@@ -177,11 +189,19 @@ public class HubEndPoint<THub, TClient> : StreamingEndPoint, IInvocationBinder w
                     _logger.LogError("Unknown hub method '{method}'", invocationDescriptor.Method);
                 }
 
-                await invocationAdapter.WriteMessageAsync(result, stream);
+                // TODO: Pool memory
+                var outStream = new MemoryStream();
+                await invocationAdapter.WriteMessageAsync(result, outStream);
+
+                var buffer = ReadableBuffer.Create(outStream.ToArray()).Preserve();
+                if (await connection.Transport.Output.WaitToWriteAsync())
```

Comment:
What happens if WaitToWriteAsync returns false? Does this mean that the channel is being closed and WaitToReadAsync() is guranteed to return false so we exit the loop? Otherwise we will never send the message....